packages/labs/gen-wrapper-vue/goldens/test-element-a/src/ElementA.vue

```diff
@@ -0,0 +1,27 @@
+<script setup lang="ts">
+import {h, useSlots} from 'vue';
+import {wrapSlots, Slots} from '@lit-labs/vue-utils/wrapper-utils.js';
+import '@lit-internal/test-element-a/element-a.js';
+
+const props = defineProps<{
+  foo?: string | undefined;
```

Comment:
Is the main point of generating a .vue file rather than a .ts file this declaration here? Because you can definitely just cast the props declaration in a .ts file to the property type and then It think not have to run the Vue compiler at all. That would make generating elements more efficient.

Reply:
Reasons to generate an SFC:
1. Can more easily type the properties and events. Vue has a JS runtime typing system which they auto-produce only when generating via an SFC. If we use regular TS, we have to manually do that and that means supporting their bespoke format.
2. Vue seems to be putting more and more features into SFCs and using them is the generally recommended path. Therefore our source is more canonical / normal this way. If we choose to produce a template using their syntax in the future, for example, we'd need this. 
3. We need to run `vue-tsc` during testing to be able to type check our output in a Vue template so we must depend on their toolchain anyway.

Reply:
1. (1) should mostly be taken care of by casting types as documented here: https://vuejs.org/guide/typescript/options-api.html#typing-component-props. Runtime times are a wrinkle, so this is the crux of the ROI question, but the core [TS type -> runtime type inference](https://github.com/vuejs/core/blob/583b6259870211c32efee0bb4a60b342799d80f7/packages/compiler-sfc/src/compileScript.ts#L1763) in the Vue compiler looks very straightforward. Essentially the name types (like `string`, `number`, `symbol`, etc.) are mapped, tuples and unions mapped to arrays, and most of the rest just becomes `Object`. This `inferRuntimeType` function appears to be the critical thing we're buying with the entire Vite toolchain. 
2. Most of their docs seem to me to cover the plain JS APIs of createApp() and createComponent(). SFC's have a section in "Scaling Up". It doesn't seem likely at all that they will move away from non-SFC components. We don't need to produce complicated templates that I can see, but the `h()` technique would probably carry us for a while anyway.
3. We may need `vue-tsc` for our own tests, but we don't need it for the generator itself, right? This gets to my big concern here: that projects that don't otherwise use Vue would have to take on much larger dependencies to publish Vue components. This is true of Angular too, but I'd love to find a way to avoid the Angular toolchain like how it looks to be possible for Vue. At the very least, rather than use Vite which includes a lot of other non-Vue tools, I'd like to see us just use the SFC compiler if really necessary: https://github.com/vuejs/vue-component-compiler/

Reply:
Yeah, I think it's fair that it mostly comes down to the ROI of point 1. Basically the entire value of having a Vue wrapper at all is type support and given the complexity of their system and it's potential to evolve, I'm pretty hesitant to reverse engineer parts of it (even if it isn't super complex atm). I also think it's valuable to generate a wrapper that has a setup that's familiar to a Vue user and using the standard toolchain is a big part of that.

The concern about deps size seems like something we should probably discuss more holistically with respect to generating wrappers. Based on that discussion we can probably more easily resolve the ROI question here.